app/src/main/java/cm/aptoide/pt/install/InstallAppSizeValidator.kt

```diff
@@ -3,12 +3,17 @@ package cm.aptoide.pt.install
 import android.os.Build
 import android.os.Environment
 import android.os.StatFs
+import cm.aptoide.pt.database.realm.Download
+import cm.aptoide.pt.utils.FileUtils
 
-class InstallAppSizeValidator {
+class InstallAppSizeValidator(val filePathProvider: FilePathProvider) {
 
-  fun hasEnoughSpaceToInstallApp(downloadSize: Long): Boolean {
-    val availableSpace = getAvailableSpace()
-    return downloadSize <= availableSpace
+  fun hasEnoughSpaceToInstallApp(download: Download): Boolean {
+    if (isAppAlreadyDownloaded(download)) {
+      return true
+    } else {
+      return download.size <= getAvailableSpace()
+    }
```

Comment:
my logaf here is low but you could do something with the ? operator here like return a one liner, up to you tho.

Reply:
the ?: in kotlin is only for null check

"The Elvis operator in Kotlin is an operator that receives two inputs and returns the first argument if it is non-null or the second one otherwise. It is fancily called the null-coalescing operator. It is a variant of the ternary operator but for null-safety checking."